de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/controller/ProjectExplorerToolBarController.java

```diff
@@ -0,0 +1,192 @@
+				searchButton.setGraphic(closeSearchImage);
+				searchButton.setTooltip(closeSearchTooltip);
+				bar.getItems().add(1, search);
+				search.requestFocus();
+			} else {
+				searchButton.setGraphic(searchImage);
+				searchButton.setTooltip(searchTooltip);
+				search.setText("");
+				bar.getItems().remove(search);
+				services.eventBroker.send(E4CEventTable.EVENT_FILTER_CHANGED, null);
+			}
+
+		});
+		bar.getItems().add(searchButton);
+
+		bar.getItems().add(new Separator());
+
+		// New Folder
+		btnNewFolder = createToolbarButton("Create New Folder", FileTable.NEWFOLDER_PNG, actionEvent -> {
+			NewFolderHandler handler = new NewFolderHandler();
+			if (handler.canExecute(services.rcpSelectionService)) {
+				handler.execute(context, services, shell);
+			}
+		});
+
+		// File Actions
+		btnNewFile = createToolbarButton("Create new File", FileTable.NEWFILE_PNG, actionEvent -> {
+			NewFileHandler handler = new NewFileHandler();
+			if (handler.canExecute(services.rcpSelectionService)) {
+				handler.execute(context, services, shell);
+			}
+		});
+
+		// Import Files
+		btnImportFiles = createToolbarButton("Import Files", FileTable.FILE_PNG, actionEvent -> {
+			FileImportHandler handler = new FileImportHandler();
+			if (handler.canExecute(services.rcpSelectionService)) {
+				handler.execute(context, services, shell);
+			}
+		});
+
+		// Delete
+		btnDelete = createToolbarButton("Delete Selected Entries", FileTable.DELETE_PNG, actionEvent -> {
+			RemoveFileCommand handler = new RemoveFileCommand();
+			if (handler.canExecute(services.rcpSelectionService)) {
+				handler.execute(context, services, shell);
+			}
+		});
+
+		// Show in Explorer
+		btnShowExplorer = createToolbarButton("Show in Explorer", FileTable.EXPLORER_PNG, actionEvent -> {
+			ShowInExplorerHandler handler = new ShowInExplorerHandler();
+			if (handler.canExecute(services.rcpSelectionService)) {
+				handler.execute(context, services, shell);
+			}
+		});
+
+		// Switch between hierarchical and flat view
+		ImageView flatViewImage = services.imageService.getFXImage(null, FileTable.FLAT_VIEW_PNG);
+		ImageView hierViewImage = services.imageService.getFXImage(null, FileTable.HIERARCICAL_VIEW_PNG);
+		Button btnToggleView = new Button("", flatViewImage);
+		btnToggleView.setTooltip(new Tooltip("Change Representation"));
+		btnToggleView.setOnAction(actionEvent -> {
+			isFlatView = !isFlatView;
+			services.eventBroker.send(E4CEventTable.EVENT_VIEW_TOGGLE, null);
+
+			if (isFlatView) {
+				btnToggleView.setGraphic(hierViewImage);
+			} else {
+				btnToggleView.setGraphic(flatViewImage);
+			}
+		});
+		bar.getItems().add(btnToggleView);
+
+		update();
+	}
+
+	/**
+	 * Creates an FX Button and adds it to the toolbar.
+	 * 
+	 * @param tooltip   The Tooltip to be displayed
+	 * @param imagePath Path to a 16px Image, ideally via FileTable
+	 * @param value     The Action to be performed once the button is clicked
+	 * @return The created button
+	 */
```

Comment:
Kann man in eine Factory/Builder Klasse auslagern siehe: de.tu_bs.cs.isf.e4cf.core.gui.java_fx.util.JavaFXBuilder